src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\AnalyticsBundle\EventListener;
+
+use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
+use SWP\Bundle\AnalyticsBundle\Repository\RequestMetricRepository;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Stopwatch\Stopwatch;
+use Symfony\Component\Stopwatch\StopwatchEvent;
+
+class KernelListener
+{
+    /**
+     * @var Stopwatch
+     */
+    protected $stopwatch;
+
+    /**
+     * @var StopwatchEvent
+     */
+    protected $stopwatchEvent;
+
+    /**
+     * @var RequestMetricRepository
+     */
+    protected $requestMetricRepository;
+
+    public function setRequestMetricRepository(RequestMetricRepository $requestMetricRepository)
+    {
+        $this->requestMetricRepository = $requestMetricRepository;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $route = $event->getRequest()->get('_route');
+
+        // Todo: check for configured routes
+        // Ignore web debug toolbar
+        if (null !== $route && '_wdt' !== $route) {
+            $this->stopwatch = new Stopwatch();
+            $this->stopwatch->start($event->getRequest()->getUri());
+        }
+    }
+
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        if (null !== $this->stopwatch) {
+            return;
+        }
+
+        $requestUri = $event->getRequest()->getUri();
+
+        // find out if this is a profiler request, if not don't analyze it
+        try {
+            $this->stopwatchEvent = $this->stopwatch->stop($requestUri);
+        } catch (\LogicException $e) {
+            // do nothing, we shouldn't be analyzing this controller
+            return false;
+        }
+
+        $duration = $this->stopwatchEvent->getDuration();
+        $route = $event->getRequest()->get('_route');
+
+        // Route is null here in test environment
```

Comment:
Well, I just wanted to avoid filling the database with junk in the dev
environment. Anyway, if we are configuring routes to be analysed, this line
will become redundant.

On Mon, Jun 13, 2016 at 12:36 PM, Daniel Read <daniel.read@sourcefabric.org>
wrote:

> This is only a problem in the test environment, as far as I can tell. The
> parameter is then set in the kernel.request event, but not in the
> kernel.response event.
>
> On Mon, Jun 13, 2016 at 12:28 PM, Paweł Mikołajczuk <
> notifications@github.com> wrote:
>
>> In src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php
>> <https://github.com/superdesk/web-publisher/pull/78#discussion_r66770282>
>> :
>>
>> > +        }
>> > +
>> > +        $requestUri = $event->getRequest()->getUri();
>> > +
>> > +        // find out if this is a profiler request, if not don't analyze it
>> > +        try {
>> > +            $this->stopwatchEvent = $this->stopwatch->stop($requestUri);
>> > +        } catch (\LogicException $e) {
>> > +            // do nothing, we shouldn't be analyzing this controller
>> > +            return false;
>> > +        }
>> > +
>> > +        $duration = $this->stopwatchEvent->getDuration();
>> > +        $route = $event->getRequest()->get('_route');
>> > +
>> > +        // Route is null here in test environment
>>
>> Why it's like that? If there can be a request without route then better
>> would be to move that check also to top. There is no point to play with
>> other things when at end you can't save it.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/superdesk/web-publisher/pull/78/files/5fbb2cfc7cad6a7bafa68838427244bcec0eb8fa#r66770282>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ARBtFWTeCXyGi8hekbmNcoGVTYt-nOkkks5qLTDPgaJpZM4Ix7A->
>> .
>>
>
>